testing/munin/Makefile.config

```diff
@@ -0,0 +1,172 @@
+OSTYPE     := $(shell uname | LANG=C tr '[A-Z]' '[a-z]' | cut -f 1 -d _)
+
+# How to figure out the hostname. (Only used in default configuration
+# files)
+HOSTNAME   := localhost
+
+# What is the safest way to create a tempfile.
+# Default is to figure it out by testing various methods.
+# Replace this with a known platform-specific method
+MKTEMP     := $(shell ./test-mktemp)
+
+# Munin version number.
+VERSION    := $(shell ./getversion)
+
+# User to run munin as
+USER       := munin
+GROUP      := munin
+
+# Default user to run the plugins as
+PLUGINUSER := nobody
+
+# Default user to run the cgi as
+CGIUSER := nobody
+
+# Which command to use to check if the USER and GROUP to run Munin as, exists.
+# These will work on most modern OSes:
+#
+GETENT     := /bin/true
+CHECKUSER  := /bin/true
+CHECKGROUP := /bin/true
+
+# For OSX, comment out the previous two lines and comment in these
+#
+#CHECKUSER  := $(shell nicl . -read /users/$(USER) >/dev/null 2>/dev/null || (echo "echo User $(USER) nonexistent. Create the user and retry; exit 2"))
+#CHECKGROUP := $(shell nicl . -read /groups/$(GROUP) >/dev/null 2>/dev/null || (echo "echo Group $(GROUP) nonexistent. Create the group and retry; exit 2"))
+
+# For OSX 10.5 (Leopard), use the following two lines instead of what's above
+#
+#CHECKUSER  := $(shell dscl . -read /Users/$(USER) >/dev/null 2>/dev/null || (echo "echo User $(USER) nonexistent. Create the user and retry; exit 2"))
+#CHECKGROUP := $(shell dscl . -read /Groups/$(GROUP) >/dev/null 2>/dev/null || (echo "echo Group $(GROUP) nonexistent. Create the group and retry; exit 2"))
+
+
+# For HP-UX, use these instead:
+#
+#CHECKUSER  := $(shell pwget -n $(USER) >/dev/null 2>/dev/null || (echo "echo User $(USER) nonexistent. Create the user and retry; exit 2"))
+#CHECKGROUP := $(shell grget -n $(GROUP) >/dev/null 2>/dev/null || (echo "echo Group $(GROUP) nonexistent. Create the group and retry; exit 2"))
+
+# For Cygwin, use these instead:
+#CHECKUSER  := $(shell id $(USER) >/dev/null 2>/dev/null || (echo "echo User $(USER) nonexistent. Create the user and retry; exit 2"))
+#CHECKGROUP := $(shell grep ^$(GROUP): /etc/group >/dev/null 2>/dev/null || (echo "echo Group $(GROUP) nonexistent. Create the group and retry; exit 2"))
+
+CHOWN      := true
+CHMOD      := chmod
+CHGRP      := true
+
+# Java compiler stuff - only needed on the buildhost
+JC         := javac
+JFLAGS     := -g -source 1.5 -target 1.5 -Xlint
+JAR        := jar
+
+# Check if the java compiler works
+#    Note that we defer JCVALID evaluation to runtime, 
+#    since $(JC) can be redefined later in a specific Makefile.config
+#    The core Makefile.config is then used as a Makefile.default
+JCVALID    = no
+
+# Check whether setruid functionality can be used
+HASSETR    := $(shell perl -e 'use Config; my @vars=("d_setruid", "d_setreuid", "d_setresuid"); foreach my $$var (@vars) { if ($$Config{$$var} eq "define") { print "1\n"; exit 0; } } print "0\n"; exit 0;' )
```

Comment:
It is copied from the munin source package and modified for easier distribution.